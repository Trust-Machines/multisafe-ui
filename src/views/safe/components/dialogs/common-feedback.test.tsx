import CommonTxFeedbackDialog from './common-feedback';
import {render} from '@testing-library/react';

test('1 Default Render', () => {
    const rendered = render(<CommonTxFeedbackDialog txId="0x123" title="Dialog Title"
                                                    description="Dialog description"/>);
    expect(rendered.container).toMatchSnapshot();
});

test('2 Requires confirmation', () => {
    const rendered = render(<CommonTxFeedbackDialog txId="0x123" title="Dialog Title"
                                                    description="Dialog description" requiresConfirmation/>);
    expect(rendered.container).toMatchSnapshot();
});