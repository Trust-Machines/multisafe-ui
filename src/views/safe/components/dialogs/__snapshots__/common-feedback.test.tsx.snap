// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`1 Default Render 1`] = `
<div>
  <h2
    class="MuiTypography-root MuiTypography-h6 MuiDialogTitle-root css-bdhsul-MuiTypography-root-MuiDialogTitle-root"
  >
    Dialog Title
    <button
      aria-label="close"
      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-lc2yqr-MuiButtonBase-root-MuiIconButton-root"
      tabindex="0"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="CloseIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
        />
      </svg>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </h2>
  <div
    class="MuiDialogContent-root css-ypiqx9-MuiDialogContent-root"
  >
    <div
      class="MuiBox-root css-542elh"
    >
      <div
        class="MuiDialogContentText-root MuiTypography-root MuiTypography-body1 css-qfso29-MuiTypography-root-MuiDialogContentText-root"
      >
        <div
          class="MuiBox-root css-1upilqn"
        >
          Dialog description
        </div>
        <div
          class="MuiBox-root css-0"
        >
          <a
            href="https://explorer.stacks.co/txid/0x123?chain=mainnet"
            rel="noreferrer"
            target="_blank"
          >
            View on Blockchain Explorer
          </a>
        </div>
      </div>
    </div>
  </div>
  <div
    class="MuiDialogActions-root MuiDialogActions-spacing css-hlj6pa-MuiDialogActions-root"
  >
    <button
      class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1e6y48t-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      Close
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;

exports[`2 Requires confirmation 1`] = `
<div>
  <h2
    class="MuiTypography-root MuiTypography-h6 MuiDialogTitle-root css-bdhsul-MuiTypography-root-MuiDialogTitle-root"
  >
    Dialog Title
    <button
      aria-label="close"
      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-lc2yqr-MuiButtonBase-root-MuiIconButton-root"
      tabindex="0"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="CloseIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
        />
      </svg>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </h2>
  <div
    class="MuiDialogContent-root css-ypiqx9-MuiDialogContent-root"
  >
    <div
      class="MuiBox-root css-542elh"
    >
      <div
        class="MuiDialogContentText-root MuiTypography-root MuiTypography-body1 css-qfso29-MuiTypography-root-MuiDialogContentText-root"
      >
        <div
          class="MuiBox-root css-1upilqn"
        >
          Dialog description
        </div>
        <div
          class="MuiBox-root css-1upilqn"
        >
          It will be available under Transactions section in a few minutes for other owners' approvals.
        </div>
        <div
          class="MuiBox-root css-0"
        >
          <a
            href="https://explorer.stacks.co/txid/0x123?chain=mainnet"
            rel="noreferrer"
            target="_blank"
          >
            View on Blockchain Explorer
          </a>
        </div>
      </div>
    </div>
  </div>
  <div
    class="MuiDialogActions-root MuiDialogActions-spacing css-hlj6pa-MuiDialogActions-root"
  >
    <button
      class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1e6y48t-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      Close
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;
